sway-core/src/metadata.rs

```diff
@@ -21,13 +21,15 @@ pub(crate) struct MetadataManager {
     md_storage_key_cache: HashMap<MetadataIndex, u64>,
     md_inline_cache: HashMap<MetadataIndex, Inline>,
     md_test_decl_index_cache: HashMap<MetadataIndex, usize>,
+    md_config_const_name_cache: HashMap<MetadataIndex, Arc<str>>,
```

Comment:
I would use `Rc<str>` instead of `Arc`.  We don't need thread safety at this stage.

Reply:
Done